x-pack/plugins/inference/public/util/http_response_into_observable.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { map, OperatorFunction, pipe, switchMap, tap } from 'rxjs';
+import { HttpResponse } from '@kbn/core/public';
+import { InferenceTaskEvent, InferenceTaskEventType } from '../../common/tasks';
+import { createObservableFromHttpResponse } from './create_observable_from_http_response';
+import {
+  createInferenceInternalError,
+  InferenceTaskError,
+  InferenceTaskErrorEvent,
+} from '../../common/errors';
+
+export function httpResponseIntoObservable<
+  T extends InferenceTaskEvent = never
+>(): OperatorFunction<HttpResponse, T> {
+  return pipe(
+    switchMap((response) => createObservableFromHttpResponse(response)),
+    map((line): T => {
+      try {
+        return JSON.parse(line);
+      } catch (error) {
+        throw createInferenceInternalError(`Failed to parse JSON`);
+      }
+    }),
+    tap((event) => {
```

Comment:
tap is usually meant for side effects like logging, so I think it would be more idiomatic to move this to a `map` (either a new map or re-use the one above)

Reply:
Hmm, _maybe_. I'm not transforming anything in this case. There's actually an example of this in the docs: https://rxjs.dev/api/operators/tap#examples